include/astarte_bson_serializer.h

```diff
@@ -201,7 +201,7 @@ void astarte_bson_serializer_append_document(
  * @param[in] double_array an array of doubles.
  * @param[in] count the number of items stored in double_array.
  */
-void astarte_bson_serializer_append_double_array(
+astarte_err_t astarte_bson_serializer_append_double_array(
```

Comment:
Add `@return` to the docstring to describe the return type, since it has been changed.